pandas/tslib.pyx

```diff
@@ -4363,6 +4371,39 @@ def get_start_end_field(ndarray[int64_t] dtindex, object field, object freqstr=N
     raise ValueError("Field %s not supported" % field)
 
 
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def get_date_name_field(ndarray[int64_t] dtindex, object field):
+    '''
+    Given a int64-based datetime index, return array of strings of date
+    name based on requested field (e.g. weekday_name)
+    '''
+    cdef:
+        _TSObject ts
+        Py_ssize_t i, count = 0
+        ndarray[object] out
+        pandas_datetimestruct dts
+        int dow
+    
+    _dayname = np.array(
+        ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'],
+         dtype=np.str )
+
+    count = len(dtindex)
+    out = np.empty(count, dtype=object)
+
+    if field == 'weekday_name':
+        for i in range(count):
+            if dtindex[i] == NPY_NAT: out[i] = -1; continue
```

Comment:
tests go in ``tseries/tests_timeseries.py`` look where things like ``is_year_end`` properties are tested